### PR TITLE
Native resolution screenshot option

### PR DESCRIPTION
**Pull request by KosmX**
TAA (Temporal anti-aliasing) can cause blurry images when changing the screenshot resolution.
Mostly OF with shaders... (for example [BLSex ](https://www.curseforge.com/minecraft/customization/bslex-shader-bsl-edit)with some TAA option)

example

### CODE REVIEW

**Comment by KosmX** [comment]
[RPG-hud](https://www.curseforge.com/minecraft/mc-mods/rpg-hud-fabric/) is another issue, fixing it won't be too easy...
(but possible with fabric-asm early risers mod detection and with turning on the right mixin in runtime if the mod is present)
When I'll have more time I think I'll do that [(example)](https://github.com/tr7zw/FirstPersonModel/blob/master/Fabric/src/main/java/dev/tr7zw/firstperson/fabric/fix/InitCompatibilityMixins.java)    
  
  

Next time when you create a mod config, you can use [Autoconfig](https://www.curseforge.com/minecraft/mc-mods/auto-config). It is much easier to set up. You don't have to do the json serializing stuff, the GUI... [(example)](https://github.com/KosmX/lock-minecart-view/blob/1.16/src/main/java/com/kosmx/lockMinecartView/LockViewConfig.java)

**Review by Minenash — COMMENTED** [review]
Thanks for the pr! I didn't know this caused an issue with optifine.

Does RPG-hud not disable itself when the hud is hidden (F1)? If not that sounds more like a bug report for them.

If my mod was big enough / had enough settings, I would use autoconfig. However, my mod is 58kb, autoconfig alone is 226 and the required cloth config 396. Which means I could go the route of making my mod 10x bigger, or require the user to also install additional mods. For such little amount of settings, I perfer the 5-15min copy-pasted 10kb (uncompressed) config compared to 500kb (compressed).

Sidenote: (And I know why this would never happen, but) I wish cloth config came bundled with modmenu so that every mod didn't have to bundle it, since practically everyone has modmenu

File: src/main/java/com/minenash/seamless_loading_screen/FinishQuit.java [hunk 1]
```diff
@@ -29,8 +30,9 @@ public static void run() {
 
         width = client.getWindow().getWidth();
         height = client.getWindow().getHeight();
```
[the remark this answers is not in the document]
Comment (KosmX) [reply src/main/java/com/minenash/seamless_loading_screen/FinishQuit.java:32]: I think, yes.
I tested it with OF and it worked for me. (If I disabled the Disable camera movement option)

**Comment by KosmX** [comment]
I was thinking about the name and the purpose of the option. There are 4K and 8K players, and the current default screenshot size is smaller than the 4k.
I think you should rename the function to "use a specific resolution", create 2 new fields for the screen width and height.

I think, most players use the same resolution every time when they play the game.  

RPG-hud. It can disable itself, except when any menu is opened.   
But this bug is not annoying, because I'll see the same after entering the world.

**Comment by KosmX** [comment]
If you want to fix it, I can help but I don't have time to do it myself.
If you want to contact me, you can find me on the Fabric discord server (KosmX#7620)

File: src/main/java/com/minenash/seamless_loading_screen/FinishQuit.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Minenash) [reply src/main/java/com/minenash/seamless_loading_screen/FinishQuit.java:32]: Those two lines and the other resize call being under the if, shouldn't actually change anything, it's just unnecessary to call it. Not sure what the camera movement has to do with it though

**Comment by Minenash** [comment]
> I was thinking about the name and the purpose of the option. There are 4K and 8K players, and the current default screenshot size is smaller than the 4k.
Technically it's "UW4K" (with a bit extra width to handle the taskbar and title bar). Though yeah now that I think about it, it definitly not enough to fully utilize a 4k 16:9 monitor.

> I think you should rename the function to "use a specific resolution", create 2 new fields for the screen width and height.
> I think, most players use the same resolution every time when they play the game.
I thought about doing that, but I ended up against it, since they'd most likely just put their screen's resolution, which wouldn't be slightly too small for most people. And at least the people I know often have mc at different sizes, mainly fullscreen, maximized, and split.

Though, I definitely can add the option to choose the resolution from a few selected ones. Like the current, one good for 4k and one good for 8k.

**Pull request merged**